### PR TITLE
Cleanup predicates.go

### PR DESCRIPTION
**Pull request by zhangxiaoyu-zidif**
**What this PR does / why we need it**:
cleanup some comments and errors.New().

**Special notes for your reviewer**:
/cc @jayunit100
 
**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go [hunk 1]
```diff
@@ -1008,7 +1007,7 @@ func (c *PodAffinityChecker) InterPodAffinityMatches(pod *v1.Pod, meta interface
 // TODO: Do we really need any pod matching, or all pods matching? I think the latter.
 func (c *PodAffinityChecker) anyPodMatchesPodAffinityTerm(pod *v1.Pod, allPods []*v1.Pod, node *v1.Node, term *v1.PodAffinityTerm) (bool, bool, error) {
 	if len(term.TopologyKey) == 0 {
-		return false, false, errors.New("Empty topologyKey is not allowed except for PreferredDuringScheduling pod anti-affinity")
+		return false, false, fmt.Errorf("Empty topologyKey is not allowed except for PreferredDuringScheduling pod anti-affinity")
```
Comment (xiang90) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: empty topologyKey ...

error message should not be capitalized.

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go [hunk 2]
```diff
@@ -1248,7 +1247,7 @@ func PodToleratesNodeTaints(pod *v1.Pod, meta interface{}, nodeInfo *schedulerca
 	return false, []algorithm.PredicateFailureReason{ErrTaintsTolerationsNotMatch}, nil
 }
 
-// Determine if a pod is scheduled with best-effort QoS
+// Check if pod is BestEffort pod
```
Comment (xiang90) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: please follow go standard for func comments:

isPodBestEffort checks if pod is scheduled with best-effort QoS

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go (hunk 1 above)
Comment (xiang90) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: also why do you change to use fmt.Errorf?

Comment (zhangxiaoyu-zidif) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: fmt.Errorf almost equals errors.New.  The whole file use fmt.Errof except here. We should unify the format and style.
As to "capitalized", I am not sure. Although it shoud be low-case in error message, it seems not the style of k8s.

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go (hunk 2 above)
Comment (zhangxiaoyu-zidif) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: Here you are right. I will fix it.

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go (hunk 1 above)
Comment (xiang90) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: > Although it shoud be low-case in error message, it seems not the style of k8s.

k8s follows go style. please fix other places when you see them. thanks.

File: plugin/pkg/scheduler/algorithm/predicates/predicates.go [hunk 3]
```diff
@@ -1248,7 +1247,7 @@ func PodToleratesNodeTaints(pod *v1.Pod, meta interface{}, nodeInfo *schedulerca
 	return false, []algorithm.PredicateFailureReason{ErrTaintsTolerationsNotMatch}, nil
 }
 
-// Determine if a pod is scheduled with best-effort QoS
+// isPodBestEffort checks if pod is scheduled with best-effort QoS
```
Comment (bsalamat) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: There are a few other functions in the file that do not follow the go style. Could you please fix those as well?

Comment (gyliu513) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: Do we need update here? Seems the godoc will not show those functions.

Comment (zhangxiaoyu-zidif) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: image
@jayunit100 thinks it is necessary, so do I. Thanks.

Comment (xiang90) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: @gyliu513 even if this is not a public api, we'd better still follow the go style.

Comment (gyliu513) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: Please note there is a PR covering the godoc for all of the public api here https://github.com/kubernetes/kubernetes/pull/47052

Comment (zhangxiaoyu-zidif) [inline plugin/pkg/scheduler/algorithm/predicates/predicates.go]: OK, good job. I will avoid conflicts. =)

**Comment by bsalamat** [comment]
/lgtm

Given #47052 will clean-up more comments.

**Comment by gyliu513** [comment]
@k8s-bot ok to test

**Pull request merged**